[#5268] Allow mapping children to zgw apis registration

vaszig opened the pull request:
Closes #5268 partly

**Changes**


- Added children rol type selection in the ZGW-APIs registration options
- Refactored registration process in order to adapt the custom children component values
- Updated the fixtures for Open Zaak (VCR tests)


**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Dockerfile/scripts

  - [x] Updated the Dockerfile with the necessary scripts from the `./bin` folder

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

SonnyBA commented on src/openforms/js/components/admin/form_design/registrations/zgw/fields/RoltypeFields.js:
```diff
@@ -133,7 +122,35 @@ const RoltypeFields = ({catalogueUrl = ''}) => {
+          }
+          noManageChildProps
+        >
+          <RoltypeSelect
```
Should both of the `RolTypeSelect` fields have the same `roleTypes`?

vaszig replied:
The `rolTypes` value is coming from the API call we make a bit above. The call is the same for all the components because it's based on the `zgwApiGroup`, `catalogueUrl`, and `caseTypeIdentification`. We already had this here for the `medewerkerRoltype` and the `partnersRoltype` so I guess there is no change. 

We had discussed this last time as well:
https://github.com/open-formulieren/open-forms/pull/5386#discussion_r2190183400

SonnyBA reviewed (commented):
Some questions and a couple of suggestions. Like discussed though it would be good if someone else took some time to also review this who is a bit more familiar with how the changes should work.

vaszig commented on src/openforms/js/compiled-lang/en.json:
tbh I usually run the compile script as well and this is what I remember. Untranslated code can be found by the https://github.com/open-formulieren/open-forms/blob/master/bin/find_untranslated_js.py script.

vaszig commented:
Apart from processing your feedback I have updated the plugin to send only the selected (when `enableSelection` is `True`) children and not all of them as I was doing and added more test cases. This was discussed with Sergei in PR 5608.

vaszig commented on src/openforms/registrations/contrib/zgw_apis/tests/test_backend.py:
```diff
@@ -2494,3 +2543,543 @@ def test_submission_with_partners_component(self, m, n):
+        assert isinstance(children_data[1], dict)
+        children_data[0].update(
+            {
+                "selected": True,
```
No, if `enableSelection` is set to `False` then the `selected` property will be null/None. Maybe this is more helpful than my explanation

https://github.com/open-formulieren/open-forms-sdk/blob/main/src/formio/components/Children/Children.jsx#L66

sergei-maertens reviewed (changes requested):
Just some minor re-shuffling of the configuration fields in the admin UI and then this is good to go!

vaszig commented on src/openforms/js/components/admin/form_design/registrations/zgw/fields/FamilyMembersDescription.js:
```diff
@@ -32,4 +33,31 @@ const PartnersDescription = () => {
+          />
+        }
+      >
+        <TextInput id="id_childrenDescription" {...fieldProps} />
```
They are linked indeed.
